status/health/periodic/source.go

```diff
@@ -20,90 +20,142 @@ import (
 	"sync"
 	"time"
 
-	conjurehealth "github.com/palantir/witchcraft-go-server/conjure/witchcraft/api/health"
+	"github.com/palantir/witchcraft-go-logging/wlog/wapp"
+	"github.com/palantir/witchcraft-go-server/conjure/witchcraft/api/health"
 	"github.com/palantir/witchcraft-go-server/status"
-	"github.com/palantir/witchcraft-go-server/status/health"
 )
 
+type Source interface {
+	Check(ctx context.Context) (state health.HealthState, params map[string]interface{})
```

Comment:
@andybradshaw I'd like your feedback particularly on this interface and whether it addresses #72

Reply:
Also should this just be a function type?
```go
type Source func(ctx context.Context) (state health.HealthState, params map[string]interface{})
```